libuserhelper.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserved.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "libuserhelper.h"
+
+#include <libuser/user.h>
+
+#include <QDebug>
+
+LibUserHelper::LibUserHelper()
+{
+}
+
+uint LibUserHelper::addGroup(const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, lu_prompt_console, NULL, &error);
+
+    if (!context) {
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return 0;
+    }
+
+    uint rv = 0;
+    struct lu_ent *ent_group = lu_ent_new();
+    if (lu_group_default(context, group.toUtf8(), false, ent_group)) {
+        if (lu_group_add(context, ent_group, &error)) {
+            rv = lu_ent_get_first_id(ent_group, LU_GIDNUMBER);
+            if (rv == LU_VALUE_INVALID_ID) {
+                qWarning() << "Invalid group id";
+                rv = 0;
+            }
+        } else {
+            qWarning() << "Adding group failed:" << lu_strerror(error);
+            lu_error_free(&error);
+        }
+    } else {
+        qWarning() << "Getting group defaults failed";
+    }
+
+    lu_ent_free(ent_group);
+    lu_end(context);
+
+    return rv;
+}
+
+bool LibUserHelper::addUserToGroup(const QString &user, const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, lu_prompt_console, NULL, &error);
```

Comment:
This `lu_prompt_console` looks scary in a daemon, which probably shouldn't even have a controlling terminal; is it possible to use NULL?

Reply:
Nulled, didn't seem to print much debugging anyway.